tests/test_callbacks.py

```diff
@@ -123,6 +125,33 @@ def test_eval_callback_vec_env():
     assert eval_callback.last_mean_reward == 100.0
 
 
+class AlwaysFailCallback(BaseCallback):
+    def __init__(self, *args, callback_false_value, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.callback_false_value = callback_false_value
+
+    def _on_step(self) -> bool:
+        return self.callback_false_value
+
+
+@pytest.mark.parametrize("model_class", [A2C, PPO, SAC, TD3, DQN, DDPG])
+@pytest.mark.parametrize("callback_false_value", [False, np.bool_(0), th.tensor(0, dtype=th.bool)])
+def test_callbacks_can_cancel_runs(model_class, callback_false_value):
+    assert not callback_false_value  # Sanity check to ensure parametrized values are valid
+    env_id = select_env(model_class)
+    model = model_class("MlpPolicy", env_id, policy_kwargs=dict(net_arch=[32]))
+    eval_callback = EvalCallback(
+        gym.make(env_id),
+        callback_after_eval=AlwaysFailCallback(callback_false_value=callback_false_value),
+        n_eval_episodes=1,
+        eval_freq=1,
+        warn=False,
+    )
+    model.learn(10, callback=eval_callback)
+
+    assert eval_callback.n_calls == 1
+
+
```

Comment:
```suggestion

```

Reply:
You can drop this, the `_on_step` is only supposed to return `bool`, see https://github.com/DLR-RM/stable-baselines3/issues/1706#issuecomment-1745376573

Reply:
I don't see the problem with having tests for edge-cases, or at the very least just for `False`, for regression purposes.

Reply:
@araffin what is your opinion?

Reply:
maybe do the check with A2C only, as the rest of the algorithm would require special parameters to avoid long runtime.
`np.bool_` seems to be a dtype, `np.array(1) < 0` would probably be a better candidate

Reply:
> maybe do the check with A2C only, as the rest of the algorithm would require special parameters to avoid long runtime.

Maybe one on-policy (A2C) and one off-policy (SAC)?

> np.bool_ seems to be a dtype, np.array(1) < 0 would probably be a better candidate

`np.array(1) < 0` evaluates to `np.bool_(0)` (`np.bool_(0) is (np.array(1)<0)` gives `True`)